src/zenml/utils/docker_utils.py

```diff
@@ -321,7 +344,12 @@ def is_local_image(image_name: str) -> bool:
     Returns:
         `True` if the image was pulled from a registry, `False` otherwise.
     """
-    docker_client = DockerClient.from_env()
+    try:
+        docker_client = DockerClient.from_env()
+    except DockerException as e:
+        raise RuntimeError(
+            "Could not create a Docker client from the environment. Is your Docker daemon running?"
+        ) from e
```

Comment:
I suggest you wrap this logic into utility method in `src/zenml/utils/docker_utils.py`, something like:
```python
def _try_get_docker_client_from_env():
    try:
        return DockerClient.from_env()
    except DockerException as e:
        raise RuntimeError(
            "Could not create a Docker client from the environment. Is your Docker daemon running?"
        ) from e
```
and use it in all other place to safe some LOC.